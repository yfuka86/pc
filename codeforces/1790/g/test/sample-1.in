6
8 10
2 4
7 8
2 4 5 6
1 2
2 3
2 4
3 4
3 5
4 6
5 6
5 7
6 8
7 8


5 4
1 1
5
3
1 2
2 3
3 4
4 5


2 1
1 0
2


1 2


4 3
1 2
2
3 4
1 2
2 3
2 4


5 4
3 2
5 3 4
2 4
1 2
2 3
3 4
4 5


1 0
1 1
1
1
