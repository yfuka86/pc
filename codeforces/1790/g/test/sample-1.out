YES
NO
YES
YES
YES
YES
